samples/QuickstartWeatherServer/Tools/HttpClientExt.cs

```diff
@@ -0,0 +1,14 @@
+
+using System.Text.Json;
+
+namespace ModelContextProtocol;
+
+internal static class HttpClientExtensions
+{
+    public static async Task<JsonDocument> ReadJsonDocument(this HttpClient client, string requestUri)
```

Comment:
@eiriktsarpalis This feels like something that should be in the SDK. Or at least something similar. A way to get either a JsonDocument or JsonElement out of an HttpClient without going through the reflection serializer.

Reply:
Don't the existing methods expose an overload that accept `JsonTypeInfo<T>`. You could pass that a source generated `JsonTypeInfo<JsonDocument>`, although I doubt it would require less code than your current approach.